packages/datadog-instrumentations/src/connect.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const shimmer = require('../../datadog-shimmer')
+const { addHook, channel, AsyncResource } = require('./helpers/instrument')
+
+const enterChannel = channel('apm:connect:middleware:enter')
+const errorChannel = channel('apm:connect:middleware:error')
+const exitChannel = channel('apm:connect:middleware:exit')
+const handleChannel = channel('apm:connect:request:handle')
+
+function wrapConnect (connect) {
+  if (typeof connect !== 'function') return connect
+
+  return function connectWithTrace () {
```

Comment:
Non-blocking Comment: Here and elsewhere, you use `WithTrace`, when we generally want to keep the notion of tracing outside of `datadog-instrumentations`.

Reply:
Completely missed the name. I think it's fine for now since it will be renamed by shimmer so it shouldn't show up in stack traces.